src/sage/combinat/nu_tamari_lattice.py

```diff
@@ -100,3 +111,142 @@ def NuTamariLattice(nu):
+
+    The vector `\delta = (\delta_1, \dots, \delta_n)` is an increment vector with respect to the path `\nu`, that is to say `\delta_i \leq \nu_i`, where `\nu_i` is the number of East steps following the `i`-th North step of `\nu`.
+
+    INPUT:
+
+    - ``p`` -- a `\nu`-Dyck word
+
+    - ``k`` -- an integer between `0` and `p.length()-1`
+
+    - ``delta`` -- a list of nonnegative integers of length p.height()
+
+    OUTPUT:
+
+    - a `\nu`-Dyck word
+
+    EXAMPLES::
+
+        sage: from sage.combinat.nu_tamari_lattice import delta_swap
+        sage: delta_swap(NuDyckWord('0101', '0101'), 3, delta = [1, 0])
+        [0, 1, 1, 0]
+        sage: delta_swap(NuDyckWord('1001110100', '0100010111'), 3, [3, 1, 0, 0, 0])
+        [1, 0, 1, 1, 1, 0, 0, 1, 0, 0]
+        sage: delta_swap(NuDyckWord('10100101000', '01001000110'), 2, [2, 3, 0, 1])
+        [1, 1, 0, 0, 1, 0, 0, 1, 0, 0, 0]
+        sage: delta_swap(NuDyckWord('10100101000', '01001000110'), 2, [1, 1, 0, 0])
+        [1, 1, 0, 0, 0, 1, 0, 1, 0, 0, 0]
+
+
+    TESTS::
+
+        sage: delta_swap(NuDyckWord('10011101000', '01000101110'), 0, [3, 1, 0, 0, 1])
+        Traceback (most recent call last):
+        ...
+        ValueError: there is no such covering move
+        sage: delta_swap(NuDyckWord('10011101000', '01000101110'), 1, [3, 1, 0, 0, 1])
+        Traceback (most recent call last):
+        ...
+        ValueError: there is no such covering move
+        sage: delta_swap(NuDyckWord('10011101000', '01000101110'), 11, [3, 1, 0, 0, 1])
+        Traceback (most recent call last):
+        ...
+        ValueError: the index is greater than the length of the path
+    """
+    if k >= p.length():
+        raise ValueError("the index is greater than the length of the path")
+    # if delta is None:
+    #     delta = [len(_) for _ in str(p._nu).split(sep='1')[1:]]
+    if k == 0 or p[k-1] == 1:
+        raise ValueError("there is no such covering move")
+    found = False
+    i = p[:k].count(1)
+    j = k
+    alt = 0
+    while not found and j <= p.length()-1:
+        if p[j]:
+            alt += delta[i]
+            i += 1
+        else:
+            alt -= 1
+        if alt == 0:
+            found = True
+        j += 1
+    q = p[:k-1] + p[k:j] + [p[k-1]] + p[j:]
+    return NuDyckWord(q, p._nu)
+
+def AltNuTamariLattice(nu, delta=None):
+    r"""
+    Return the `(\delta,\nu)`-Tamari lattice (or alt `\nu`-Tamari lattice).
+
+    For more information, see [CC2023]_.
+
+    The path `\nu` is a path of North steps (represented as `1` s) and East steps (represented as `0` s).
+
+    The vector `\delta = (\delta_1, \dots, \delta_n)` is an increment vector with respect to the path `\nu`, that is to say `\delta_i \leq \nu_i`, where `\nu_i` is the number of `0` s following the `i`-th `1` of `\nu`. If not provided, `\delta` is set by default to produce the classical `\nu`-Tamari lattice.
+
```

Comment:
break at 80 char

Reply:
I have broken most lines at 80 characters, except some lines in the actual code. It should be better now!